### ISSUE TITLE
Clippy errors in build.rs suppress errors in the crate itself

### ISSUE THREAD

**Issue by Wilfred** [rust-lang/rust-clippy#9534]
### Summary

When there are clippy errors in build.rs, `cargo clippy` no longer reports any errors in the rest of the crate.

This is confusing when using `rust-analyzer` with clippy as the check command. If there are any clippy errors in build.rs, you lose all red underlines in the other files you're editing.

### Reproducer

I had a main.rs with a type error

```rust
fn main() {
    return 1;
}
```

and a build.rs with a clippy error:

```
fn say_hello(s: &str) {
    if s.contains("unix") {
        println!("unixish");
    } else if s.contains("macos") {
        println!("unixish");
    } else {
        println!("probably windows");
    }
}

fn main() {
    say_hello("abc");
}
```

I expected to see this happen:

```
$ cargo clippy
... reports the rustc failure in main.rs ...
```

Instead, this happened:

```
$ cargo clippy
... only complains about build.rs
```

### Version

```text
rustc 1.63.0 (4b91a6ea7 2022-08-08)
binary: rustc
commit-hash: 4b91a6ea7258a947e59c6522cd5898e7c0a6a88f
commit-date: 2022-08-08
host: x86_64-unknown-linux-gnu
release: 1.63.0
LLVM version: 14.0.5
```

**Comment by Alexendoo** [comment]
That would be confusing indeed. As a workaround you could downgrade the errors to warnings just for the build script, `#![warn(clippy::correctness)]`
